File: Dynamic Programming/canConstruct.py
def canConstruct(target,wordBank):
    if target == '':
        return True
    
    for word in wordBank:
        if target.startswith(word):
            suffix = target[len(word):]
            if canConstruct(suffix,wordBank) is True:
                return True
            
    return False
    
print(canConstruct("abcdef",["ab","abc","cd","def","abcd"]))
print(canConstruct("skateboard",["bo","rd","ate","t","ska","sk","boar"]))
print(canConstruct("enterapotentpot",["a","p","ent","enter","ot","o","t"]))
print(canConstruct("eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeef",["e","ee","eee","eeee","eeeee","eeeeee","eeeeeee"]))

File: Dynamic Programming/canConstruct.js
const canConstruct = (target,wordBank) => {
    if (target == " "){
        return true
    }

    for(let word of wordBank){
        if (target.indexOf(word) == 0){
            const suffix = target.slice(word.length)
            if(canConstruct(suffix,wordBank) === true){
                return true
            }
        }
    }
    return false
}

console.log(canConstruct("abcdef",["ab","abc","cd","def","abcd"]))
console.log(canConstruct("skateboard",["bo","rd","ate","t","ska","sk","boar"]))
console.log(canConstruct("enterapotentpot",["a","p","ent","enter","ot","o","t"]))
console.log(canConstruct("eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeef",["e","ee","eee","eeee","eeeee","eeeeee","eeeeeee"]))